crates/core/src/delta_datafusion/cdf/scan.rs

```diff
@@ -0,0 +1,93 @@
+use std::any::Any;
+use std::fmt::Formatter;
+use std::sync::Arc;
+
+use arrow_schema::{DataType, Field, SchemaRef, TimeUnit};
+use datafusion::execution::{SendableRecordBatchStream, TaskContext};
+use datafusion::physical_plan::{DisplayAs, DisplayFormatType, ExecutionPlan};
+use datafusion_physical_expr::{Partitioning, PhysicalSortExpr};
+use lazy_static::lazy_static;
+
+/// Change type column name
+pub const CHANGE_TYPE_COL: &str = "_change_type";
+/// Commit version column name
+pub const COMMIT_VERSION_COL: &str = "_commit_version";
+/// Commit Timestamp column name
+pub const COMMIT_TIMESTAMP_COL: &str = "_commit_timestamp";
+
+lazy_static! {
+    pub static ref CDC_PARTITION_SCHEMA: Vec<Field> = vec![
+        Field::new(COMMIT_VERSION_COL, DataType::Int64, true),
+        Field::new(
+            COMMIT_TIMESTAMP_COL,
+            DataType::Timestamp(TimeUnit::Millisecond, None),
+            true
+        )
+    ];
+    pub static ref ADD_PARTITION_SCHEMA: Vec<Field> = vec![
+        Field::new(CHANGE_TYPE_COL, DataType::Utf8, true),
+        Field::new(COMMIT_VERSION_COL, DataType::Int64, true),
+        Field::new(
+            COMMIT_TIMESTAMP_COL,
+            DataType::Timestamp(TimeUnit::Millisecond, None),
+            true
+        ),
+    ];
+}
+
+/// Physical execution of a scan
+#[derive(Debug, Clone)]
+pub struct DeltaCdfScan {
+    plan: Arc<dyn ExecutionPlan>,
+    schema: SchemaRef,
+}
+
+impl DeltaCdfScan {
+    /// Creates a new scan
+    pub fn new(plan: Arc<dyn ExecutionPlan>, schema: SchemaRef) -> Self {
+        Self { plan, schema }
+    }
+}
+
+impl DisplayAs for DeltaCdfScan {
+    fn fmt_as(&self, _t: DisplayFormatType, f: &mut Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl ExecutionPlan for DeltaCdfScan {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.schema.clone()
+    }
+
+    fn output_partitioning(&self) -> Partitioning {
+        self.plan.output_partitioning()
+    }
+
+    fn output_ordering(&self) -> Option<&[PhysicalSortExpr]> {
+        self.plan.output_ordering()
+    }
+
+    fn children(&self) -> Vec<Arc<dyn ExecutionPlan>> {
+        vec![self.plan.clone()]
```

Comment:
this is probably my own ignorance regarding all these apis, but all the other things (like `output_ordering` etc) seem to delegate to `self.plan`, but this one says that `plan` is the child. In particular, if this returns `plan` I would have expected `with_new_children` to _replace_ `plan`, but it instead makes the new children children of `plan`.

Anyway, just something to think of :)

Reply:
You are right here, since this is a scan, it should be an empty vec